src/token/tests.rs

```diff
@@ -9,41 +9,85 @@
  * SPDX-License-Identifier: MIT OR Apache-2.0
  */
 
-#[cfg(not(feature = "std"))]
-use alloc::{
-    string::{String, ToString},
-    vec,
-};
-
+use super::*;
+use crate::common::test_helper::MockCipher;
+use crate::error::CoseCipherError;
+use alloc::vec::Vec;
+use alloc::{string::ToString, vec};
+use base64::Engine;
 use ciborium::value::Value;
+use core::convert::Infallible;
 use coset::cwt::ClaimsSetBuilder;
-use coset::iana::{Algorithm, CoapContentFormat, CwtClaimName};
-use coset::{AsCborValue, CoseKey, CoseKeyBuilder, CoseMac0Builder, HeaderBuilder};
+use coset::iana::Algorithm::{A128GCM, A128KW, ES256, ES384, HMAC_256_256};
+use coset::iana::{Algorithm, CoapContentFormat, CwtClaimName, EllipticCurve};
+use coset::{AsCborValue, CoseKey, CoseKeyBuilder, CoseMac0Builder, HeaderBuilder, Label};
+use rand::rngs::ThreadRng;
 
-use crate::common::test_helper::{FakeCrypto, FakeRng};
-use crate::error::CoseCipherError;
+/// Generates a test key with content `[0, 1, 2, 3, 4, 5, ...]` and key id `[0xDC, 0xAF]`.
+fn example_key_one(alg: Algorithm) -> CoseKey {
+    CoseKeyBuilder::new_symmetric_key(vec![
+        0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 0xa, 0xb, 0xc, 0xd, 0xe, 0xf,
+    ])
+    .key_id(vec![0xDC, 0xAF])
+    .algorithm(alg)
+    .build()
+}
 
-use super::*;
+/// Generates a test key with content `[10, 9, 8, 7, 6, ...]` and key id `[0xCA, 0xFE]`.
```

Comment:
The content is actually `[F, E, D, C, ...]`

Reply:
```suggestion
/// Generates a test key with content `[0xF, 0xE, 0xD, 0xC, 0xB, ...]` and key id `[0xCA, 0xFE]`.
```